DuckDuckGo/DefaultNetworkProtectionVisibility.swift

```diff
@@ -0,0 +1,92 @@
+//  limitations under the License.
+//
+
+#if NETWORK_PROTECTION
+
+import Foundation
+import BrowserServicesKit
+import Waitlist
+import NetworkProtection
+import Core
+
+struct DefaultNetworkProtectionVisibility: NetworkProtectionFeatureVisibility {
+    private let privacyConfigurationManager: PrivacyConfigurationManaging
+    private let networkProtectionTokenStore: NetworkProtectionTokenStore?
+    private let networkProtectionAccessManager: NetworkProtectionAccess?
+    private let featureFlagger: FeatureFlagger
+
+    init(privacyConfigurationManager: PrivacyConfigurationManaging = ContentBlocking.shared.privacyConfigurationManager,
+         networkProtectionTokenStore: NetworkProtectionTokenStore? = NetworkProtectionKeychainTokenStore(),
+         networkProtectionAccessManager: NetworkProtectionAccess? = NetworkProtectionAccessController(),
+         featureFlagger: FeatureFlagger = AppDependencyProvider.shared.featureFlagger) {
+        self.privacyConfigurationManager = privacyConfigurationManager
+        self.networkProtectionTokenStore = networkProtectionTokenStore
+        self.networkProtectionAccessManager = networkProtectionAccessManager
+        self.featureFlagger = featureFlagger
+    }
+
+    /// A version with fewer dependencies that can't check for isWaitlistUser()
+    static func availablity() -> DefaultNetworkProtectionVisibility {
+        DefaultNetworkProtectionVisibility(networkProtectionTokenStore: nil, networkProtectionAccessManager: nil)
+    }
+
+    func isWaitlistBetaActive() -> Bool {
+        privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlistBetaActive)
+    }
+    
+    func isWaitlistUser() -> Bool {
+        guard let networkProtectionTokenStore, let networkProtectionAccessManager else {
+            preconditionFailure("networkProtectionTokenStore and networkProtectionAccessManager must be non-nil")
+        }
+
+        let hasLegacyAuthToken = {
+            guard let authToken = try? networkProtectionTokenStore.fetchToken(),
+                  !authToken.hasPrefix(NetworkProtectionKeychainTokenStore.authTokenPrefix) else {
+                return false
+            }
+            return true
+        }()
+        let hasBeenInvited = {
+            let vpnAccessType = networkProtectionAccessManager.networkProtectionAccessType()
+            return vpnAccessType == .waitlistInvited || vpnAccessType == .inviteCodeInvited
+        }()
+
+        return hasLegacyAuthToken || hasBeenInvited
+    }
+
+    // todo - https://app.asana.com/0/0/1206844038943626/f
+    func isPrivacyProLaunched() -> Bool {
+#if SUBSCRIPTION && ALPHA
+        true
+#else
+        featureFlagger.isFeatureOn(.subscription)
+#endif
+    }
+    
+    // todo - https://app.asana.com/0/0/1206844038943626/f
+    func shouldMonitorEntitlement() -> Bool {
+#if SUBSCRIPTION && ALPHA
+        true
+#else
+        isPrivacyProLaunched()
+#endif
+    }
+}
```

Comment:
Similarly, this will also be updated once the feature flags are ready. Most likely it will just be the same as `isPrivacyProLaunched()` as we should always validate the entitlement once PP launches.